test/rekt/features/apiserversource/control_plane.go

```diff
@@ -0,0 +1,126 @@
+/*
+Copyright 2021 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package apiserversource
+
+import (
+	"context"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	sourcesv1 "knative.dev/eventing/pkg/apis/sources/v1"
+	sourcesclientsetv1 "knative.dev/eventing/pkg/client/clientset/versioned/typed/sources/v1"
+	eventingclient "knative.dev/eventing/pkg/client/injection/client"
+	"knative.dev/eventing/test/rekt/resources/apiserversource"
+	"knative.dev/reconciler-test/pkg/environment"
+	"knative.dev/reconciler-test/pkg/feature"
+	"knative.dev/reconciler-test/pkg/state"
+)
+
+func ControlPlaneConformance(apiServerSourceName string) *feature.FeatureSet {
+	fs := &feature.FeatureSet{
+		Name: "Knative ApiServerSource Specification - Control Plane",
+		Features: []feature.Feature{
+			*ControlPlaneApiServerSource(apiServerSourceName),
+		},
+	}
+
+	return fs
+}
+
+func ControlPlaneApiServerSource(apiServerSourceName string) *feature.Feature {
+	f := feature.NewFeatureNamed("Conformance")
+
+	f.Setup("Set ApiServerSource Name", setApiServerSourceName(apiServerSourceName))
+
+	f.Stable("EventMode").
+		Must("ApiServerSource MUST allow retrieving the event payloads of ObjectReference or ResourceEvent",
+			eventMode()).
```

Comment:
this test does not seem to test what the spec is saying. The test confirms we can set the value, but the conformance line is talking about the shape of the event payload I believe, so your test will need to get more involved and see the event on the data plane side.